functions/Set-DbaPowerPlan.ps1

```diff
@@ -83,160 +68,68 @@ function Set-DbaPowerPlan {
-
-                            $cimInstance = Get-CimInstance -Namespace root\cimv2\power -ClassName win32_PowerPlan -Filter "ElementName = '$powerPlanRequested'" -CimSession $CIMSession
-                            if ($cimInstance) {
-                                # Because the activate method for powerplans is broken on windows server 2019, we use the powrprof.dll instead
-                                [System.Guid]$powerPlanGuid = $cimInstance.InstanceID -replace '.*{(.*)}', '$1'
-                                $scriptBlock = {
-                                    Param ($Guid)
-                                    $powerSetActiveSchemeDefinition = '[DllImport("powrprof.dll", CharSet = CharSet.Auto)] public static extern uint PowerSetActiveScheme(IntPtr RootPowerKey, Guid SchemeGuid);'
-                                    $powrprof = Add-Type -MemberDefinition $powerSetActiveSchemeDefinition -Name 'Win32PowerSetActiveScheme' -Namespace 'Win32Functions' -PassThru
-                                    $powrprof::PowerSetActiveScheme([System.IntPtr]::Zero, $Guid)
-                                }
-                                if ($IncludeCred) {
-                                    $returnCode = Invoke-CommandWithFallback -ComputerName $computer -ScriptBlock $scriptBlock -ArgumentList $powerPlanGuid -Raw -Credential $Credential
-                                } else {
-                                    $returnCode = Invoke-CommandWithFallback -ComputerName $computer -ScriptBlock $scriptBlock -ArgumentList $powerPlanGuid -Raw
-                                }
-                                if ($returnCode -ne 0) {
-                                    Stop-Function -Message "Couldn't set the requested Power Plan '$powerPlanRequested' on $computer (ReturnCode: $returnCode)." -Category ConnectionError -Target $computer
-                                    return
-                                }
-                            } else {
-                                Stop-Function -Message "Couldn't find the requested Power Plan '$powerPlanRequested' on $computer." -Category ConnectionError -Target $computer
-                                return
-                            }
-                        } else {
-                            Stop-Function -Message "Couldn't set Power Plan on $computer." -Category ConnectionError -ErrorRecord $_ -Target $computer
-                            return
+            } else {
+                if ($Pscmdlet.ShouldProcess($computer, "Changing Power Plan from $powerPlan to $powerPlanRequested")) {
+                    [System.Guid]$powerPlanGuid = $change.RecommendedInstanceId
+                    $scriptBlock = {
+                        Param ($Guid)
+                        $powerSetActiveSchemeDefinition = '[DllImport("powrprof.dll", CharSet = CharSet.Auto)] public static extern uint PowerSetActiveScheme(IntPtr RootPowerKey, Guid SchemeGuid);'
```

Comment:
awesome!

Reply:
Yes, but this code is not new but only in a different place. We uses this for some months now.

Reply:
I probably said the same thing last time I saw it 😅